grails-app/controllers/au/org/emii/netcdfsubset/SubsetController.groovy

```diff
@@ -21,13 +23,29 @@ class SubsetController {
         response.setContentType("application/octet-stream")
         response.setHeader("Content-disposition", "filename=${filenameToServe(subsetRequest)}")
 
-        try {
-            subsetService.subset(subsetRequest.typeName, subsetRequest.CQL_FILTER, response.outputStream)
+        if (isThreadAvailable()) {
```

Comment:
The only reason I had to use ``isThreadAvailable()`` is that without this if statement (when using just ``up()`` and ``down()``) any successive subset requests to an existing job will not trigger an interrupt exception and will continue to hang around in a queue in the background until a new thread is available for them. This is not behaviour we really want (as a job that takes 3 days could block up subsetter and after that's finished, the queued jobs just run automatically afterwards). I needed behaviour here such that if a job can't get a permit from the semaphore, it just dies then and there (hence if statement). But I agree with what you say above, that there is a race condition there that could cause trouble.